client/state/themes/theme-details/selectors.js

```diff
@@ -2,5 +2,21 @@
 
 export function getThemeDetails( state, id ) {
 	const theme = state.themes.themeDetails.get( id );
-	return theme ? theme.toJS() : {};
+	return theme ? formatPrice( theme.toJS() ) : {};
+}
+
+// Convert price to format used by v1.2 themes API to fit with existing components.
+// TODO (seear): remove when v1.2 theme details endpoint is added
+function formatPrice( theme ) {
+	const price = theme.price;
+	if ( price && price.value && price.currency ) {
+		theme.price = price.value.toLocaleString( 'default', {
+			style: 'currency',
+			currency: price.currency,
+			minimumFractionDigits: 0,
+		} );
+	} else if ( price ) {
```

Comment:
I still find it a bit confusing, TBH. The check is a little arbitrary -- the first branch checks for the presence of `price` and two of its threee properties, while the other also checks for `price`, and requires the presence of the third property...

Also, if the second branch is hit, we still need the `dangerousHtml` thing to view it, right?